### PR TITLE
Refactor DeepSeek model for two-batch overlap

### PR DESCRIPTION
**Pull request by fzyzcjy**
## Motivation

Extract logic into functions, which will be reused by two batch overlap.

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: python/sglang/srt/models/deepseek_v2.py [hunk 1]
```diff
@@ -393,6 +383,32 @@ def _forward_shared_experts(self, hidden_states):
         else:
             return None
 
+    def _forward_select_experts(self, forward_mode, hidden_states, router_logits):
+        if (
+            forward_mode is not None
+            and not forward_mode.is_idle()
+            and hidden_states.shape[0] > 0
+        ):
```
Comment (ch-wan) [inline python/sglang/srt/models/deepseek_v2.py]: how about using
```python
if  router_logits is not None:
```

**Review by ch-wan — APPROVED** [review]

**Comment by fzyzcjy** [comment]
I made a larger refactor today
